### PR TITLE
Add DeletePEConnection to plsClient

### PR DESCRIPTION
**Pull request by jwtty**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
When deleting a PLS, need to delete all existing PE connections first. So add this client interface.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Add azure private link service client interface to delete a private endpoint connection.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
/lgtm

File: pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient_test.go [hunk 1]
```diff
@@ -319,6 +350,13 @@ func getTestPrivateLinkService(name string) network.PrivateLinkService {
 	}
 }
 
+func getTestPrivateEndpointConnection(PLSName string, PEConnName string) network.PrivateEndpointConnection {
+	return network.PrivateEndpointConnection{
+		ID:   to.StringPtr(fmt.Sprintf("/subscriptions/subscriptionID/resourceGroups/rg/providers/"+PLSResourceType+"/%s/%s/%s", PLSName, PEConnResourceType, PEConnName)),
```
Comment (lzhecheng) [inline pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient_test.go:355]: Why not like this? Current one seems a bit strange :confused:
```suggestion
		ID:   to.StringPtr(fmt.Sprintf("/subscriptions/subscriptionID/resourceGroups/rg/providers/%s/%s/%s/%s", PLSResourceType, PLSName, PEConnResourceType, PEConnName)),
```

Comment (jwtty) [reply pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient_test.go:355]: I was following the above `getTestPrivateLinkService` function. But I don't see any problem changing it.

Comment (lzhecheng) [reply pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient_test.go:355]: I think all of they can be changed.

**Pull request merged**